src/core/llm/services/vercel.ts

```diff
@@ -38,6 +39,8 @@ export class VercelLLMService implements ILLMService {
     private sessionEventBus: SessionEventBus;
     private readonly sessionId: string;
     private toolSupportCache: Map<string, boolean> = new Map();
+    // Map of toolCallId -> queue of raw results to emit with that callId
+    private rawResultsByCallId: Map<string, any[]> = new Map();
```

Comment:
use unknown not any

or better, get the RESULT type mentioned here: https://ai-sdk.dev/docs/reference/ai-sdk-core/stream-text#tools.tool.execute